src/overlays/actors/ovl_En_Bbfall/z_en_bbfall.c

```diff
@@ -37,140 +47,717 @@ const ActorInit En_Bbfall_InitVars = {
+}
+
+void EnBbfall_Frozen(EnBbfall* this, GlobalContext* globalCtx) {
+    DECR(this->timer);
+
+    if (this->timer == 0) {
+        EnBbfall_Thaw(this, globalCtx);
+        if (this->actor.colChkInfo.health == 0) {
+            EnBbfall_SetupDead(this, globalCtx);
+        } else {
+            EnBbfall_SetupDown(this);
+        }
+    }
+}
+
+void EnBbfall_UpdateDamage(EnBbfall* this, GlobalContext* globalCtx) {
+    if (this->collider.base.acFlags & AC_HIT) {
+        this->collider.base.acFlags &= ~AC_HIT;
+        this->collider.base.atFlags &= ~(AT_HIT | AT_BOUNCED);
+        this->collider.base.atFlags &= ~AT_ON;
+        if ((this->drawDmgEffType != ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX) ||
+            (!(this->collider.elements[0].info.acHitInfo->toucher.dmgFlags & 0xDB0B3))) {
+            Actor_SetDropFlagJntSph(&this->actor, &this->collider);
+            this->flameOpacity = 0;
+            this->flameScaleY = 0.0f;
+            this->flameScaleX = 0.0f;
+            EnBbfall_DisableColliders(this);
+            EnBbfall_Thaw(this, globalCtx);
+
+            if (Actor_ApplyDamage(&this->actor) == 0) {
+                Enemy_StartFinishingBlow(globalCtx, &this->actor);
+            }
+
+            if (this->actor.colChkInfo.damageEffect == EN_BBFALL_DMGEFF_ICE_ARROW) {
+                EnBbfall_Freeze(this);
+                if (this->actor.colChkInfo.health == 0) {
+                    this->timer = 3;
+                    this->collider.base.acFlags &= ~AC_ON;
+                }
+
+                EnBbfall_SetupFrozen(this);
+            } else if (this->actor.colChkInfo.health == 0) {
+                EnBbfall_SetupDead(this, globalCtx);
+            } else {
+                EnBbfall_SetupDamage(this);
+            }
+
+            if (this->actor.colChkInfo.damageEffect == EN_BBFALL_DMGEFF_LIGHT_ARROW) {
+                this->drawDmgEffAlpha = 4.0f;
+                this->drawDmgEffScale = 0.4f;
+                this->drawDmgEffType = ACTOR_DRAW_DMGEFF_LIGHT_ORBS;
+                Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG,
+                            this->collider.elements[0].info.bumper.hitPos.x,
+                            this->collider.elements[0].info.bumper.hitPos.y,
+                            this->collider.elements[0].info.bumper.hitPos.z, 0, 0, 0, CLEAR_TAG_SMALL_LIGHT_RAYS);
+            }
+        }
+    } else {
+        if (this->collider.base.atFlags & AT_BOUNCED) {
+            this->collider.base.atFlags &= ~(AT_HIT | AT_BOUNCED);
+            EnBbfall_DisableColliders(this);
+            if (this->actionFunc != EnBbfall_Down) {
+                this->actor.world.rot.y = this->actor.yawTowardsPlayer + 0x8000;
+                this->actor.shape.rot.y = this->actor.world.rot.y;
+                EnBbfall_SetupDown(this);
+            }
+        }
+    }
+}
+
+void EnBbfall_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnBbfall* this = THIS;
+    Sphere16* sphere;
+    Vec3f sp5C;
```

Comment:
```suggestion
    Vec3f diff;
```
Looks to be the diff/distance between the the previous 2 flame positions.